networkx/algorithms/centrality/tests/test_domirank.py

```diff
@@ -0,0 +1,197 @@
+        b_answer = {
+            0: 0.7916661947954937,
+            1: 0.7916661947954933,
+            2: 0.7916661947954919,
+            3: 0.791666194795491,
+            4: 0.7916661947954975,
+        }
+        b, _, __ = nx.domirank(G, sigma=0.5)
+        b_answer = {
+            0: 0.66666579246521,
+            1: 0.66666579246521,
+            2: 0.66666579246521,
+            3: 0.66666579246521,
+            4: 0.66666579246521,
+        }
+        for n in sorted(G):
+            assert b[n] == pytest.approx(b_answer[n], abs=5e-3)
+
+    def test_P3(self):
+        """DomiRank centrality: P3"""
+        G = nx.path_graph(3)
+        b, _, __ = nx.domirank(G)
+        b_answer = {
+            0: -2.3477606773376465,
+            1: 4.496389865875244,
+            2: -2.3477606773376465,
+        }
+        for n in sorted(G):
+            assert b[n] == pytest.approx(b_answer[n], abs=5e-3)
+
+        b_answer = {0: -2.366595958759845, 1: 4.523026818398872, 2: -2.366595958759845}
+        b, _, __ = nx.domirank(G, analytical=True)
+        for n in sorted(G):
+            assert b[n] == pytest.approx(b_answer[n], abs=5e-3)
+
+        b_answer = {
+            0: 0.13808214664459229,
+            1: 0.609459638595581,
+            2: 0.13808214664459229,
+        }
+        b, _, __ = nx.domirank(G, sigma=0.5)
+        for n in sorted(G):
+            assert b[n] == pytest.approx(b_answer[n], abs=5e-3)
+
+    def test_P3_dt(self):
+        """DomiRank centrality: P3"""
+        G = nx.path_graph(3)
+        b_answer = {
+            0: -2.3666865825653076,
+            1: 4.523157596588135,
+            2: -2.3666865825653076,
+        }
+        b, _, __ = nx.domirank(G, dt=0.5)
+        for n in sorted(G):
+            assert b[n] == pytest.approx(b_answer[n], abs=5e-3)
+
+    def test_P3_eps(self):
+        """DomiRank centrality: P3"""
+        G = nx.path_graph(3)
+        b_answer = {
+            0: -2.3477983474731445,
+            1: 4.496443748474121,
+            2: -2.3477983474731445,
+        }
+        b, _, bBool = nx.domirank(G, epsilon=1e-7)
+        assert bBool == True
+        for n in sorted(G):
+            assert b[n] == pytest.approx(b_answer[n], abs=5e-3)
+
+    def test_P3_max_iter(self):
+        """DomiRank centrality: P3"""
+        G = nx.path_graph(3)
+        b_answer = {0: -2.364804744720459, 1: 4.5204758644104, 2: -2.364804744720459}
+        b, _, bBool = nx.domirank(G, max_iter=5000)
+        for n in sorted(G):
+            assert b[n] == pytest.approx(b_answer[n], abs=5e-3)
+
+    def test_P3_check_step(self):
+        """DomiRank centrality: P3"""
+        G = nx.path_graph(3)
+        b_answer = {0: -2.346951484680176, 1: 4.495236396789551, 2: -2.346951484680176}
+        b, _, bBool = nx.domirank(G, check_step=1)
+        for n in sorted(G):
+            assert b[n] == pytest.approx(b_answer[n], abs=5e-3)
+
+
+class TestDomirankDirected:
+    def __init__(self):
```

Comment:
I should add to this that I think I've seen @dschult mention (don't have an immediate reference, so take this with a grain of salt as I might have misunderstood) that NetworkX is trying to move away from the class-based test model--the `test_eigenvector_centrality.py` tests you're basing this on are nearly 6 years old--in favor of just using functions. I don't have the exact reasoning as to why this choice was made, but I figure it's worth warning you before you start working on something you might end up having to rewrite anyways.

Reply:
Noted! Yes, I just took test_eigenvector_centrality.py as a reference, as it is computationally speaking the centrality that is the most similar to domirank. But, if necessary, I will of course rework the test file to be function based rather than class based, perhaps, with a bit if your help...